src/shared/components/home/tagline-form.tsx

```diff
@@ -1,58 +1,117 @@
 import { capitalizeFirstLetter } from "@utils/helpers";
 import { Component, InfernoMouseEvent, linkEvent } from "inferno";
-import { EditSite, Tagline } from "lemmy-js-client";
-import { I18NextService } from "../../services";
+import { Tagline } from "lemmy-js-client";
+import { HttpService, I18NextService } from "../../services";
 import { Icon, Spinner } from "../common/icon";
 import { MarkdownTextArea } from "../common/markdown-textarea";
 import { tippyMixin } from "../mixins/tippy-mixin";
+import { Paginator } from "../common/paginator";
+import classNames from "classnames";
+import { isBrowser } from "@utils/browser";
+import { Prompt } from "inferno-router";
 
-interface TaglineFormProps {
-  taglines: Array<Tagline>;
-  onSaveSite(form: EditSite): void;
-  loading: boolean;
+interface EditableTagline {
+  change?: "update" | "delete" | "create";
+  editMode?: boolean;
+  tagline: Tagline;
+}
+
+function markForUpdate(editable: EditableTagline) {
+  if (editable.change !== "create") {
+    editable.change = "update";
+  }
 }
 
 interface TaglineFormState {
-  taglines: Array<string>;
-  editingRow?: number;
+  taglines: Array<EditableTagline>;
+  page: number;
+  loading: boolean;
 }
 
 @tippyMixin
-export class TaglineForm extends Component<TaglineFormProps, TaglineFormState> {
+export class TaglineForm extends Component<
+  Record<never, never>,
+  TaglineFormState
+> {
   state: TaglineFormState = {
-    editingRow: undefined,
-    taglines: this.props.taglines.map(x => x.content),
+    taglines: [],
+    page: 1,
+    loading: false,
   };
   constructor(props: any, context: any) {
     super(props, context);
+    this.handlePageChange = this.handlePageChange.bind(this);
+  }
+
+  componentWillMount(): void {
+    if (isBrowser()) {
+      this.handlePageChange(1);
+    }
+  }
+
+  hasPendingChanges(): boolean {
+    return this.state.taglines.some(x => x.change);
   }
 
   render() {
     return (
       <div className="tagline-form col-12">
+        <Prompt
+          message={I18NextService.i18n.t("block_leaving")}
+          when={this.hasPendingChanges()}
+        />
         <h1 className="h4 mb-4">{I18NextService.i18n.t("taglines")}</h1>
         <div className="table-responsive col-12">
-          <table id="taglines_table" className="table table-sm table-hover">
+          <table
+            id="taglines_table"
+            className="table table-sm table-hover align-middle"
+          >
             <thead className="pointer">
               <th></th>
+              <th style="width:60px"></th>
```

Comment:
It'd be nice at some point to do away with every table in this codebase. They're not as responsive friendly (even with side-scrolling), and editing both emojis and taglines has been a pain on mobile.

I'll make an issue for that.